crates/typst-library/src/text/smallcaps.rs

```diff
@@ -64,10 +64,32 @@ pub struct SmallcapsElem {
 impl Show for Packed<SmallcapsElem> {
     #[typst_macros::time(name = "smallcaps", span = self.span())]
     fn show(&self, _: &mut Engine, styles: StyleChain) -> SourceResult<Content> {
-        Ok(self
-            .body
-            .clone()
-            .styled(TextElem::set_smallcaps_from_caps(self.all(styles)))
-            .styled(TextElem::set_smallcaps(true)))
+        let sc = if self.all(styles) {
+            SmallcapsSetting::All
+        } else {
+            SmallcapsSetting::Minuscules
+        };
+        Ok(self.body.clone().styled(TextElem::set_smallcaps(sc)))
+    }
+}
+
+/// What becomes small capitals.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum SmallcapsSetting {
+    /// No small capitals.
+    Disabled,
```

Comment:
Disabled could be removed in favor of using `Option<..>` and just `Smallcaps` might be enough for the name.